crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -1351,7 +1351,13 @@ NamedExpression: ast::ParenthesizedExpr = {
 };
 
 LambdaDef: ast::ParenthesizedExpr = {
-    <location:@L> "lambda" <location_args:@L> <parameters:ParameterList<UntypedParameter, StarUntypedParameter, StarUntypedParameter>?> <end_location_args:@R> ":" <body:Test<"all">> <end_location:@R> =>? {
+    <location:@L> "lambda" <location_args:@L> <parameters:ParameterList<UntypedParameter, StarUntypedParameter, StarUntypedParameter>?> <end_location_args:@R> ":" <fstring_middle:fstring_middle?> <body:Test<"all">> <end_location:@R> =>? {
```

Comment:
I tried to add the `fstring_middle` optional token after the body but that creates conflict when generating. This makes sense and I guess the solution would be to modify `Test<"all">` as you suggested in Discord but I don't think it's worth the complexity right now.